test/integration/model/json.test.js

```diff
@@ -243,6 +243,22 @@ describe(Support.getTestDialectTeaser('Model'), function() {
         });
       });
 
+      it('should be possible to save/update a NULL as the field value', function () {
+        return this.Event.create({data:{a:1}}) // create with non-NULL data field
+        .then( function(event) { // set data field to null
+          event.data = null;
+          return event.save();
+        })
+        .then( function(event) { // fetch it again
+          return this.Event.findAll({id: event.id});
```

Comment:
"this" refers to the inner function and therefor does not know about the Event. You have to either bind this to the promise or add a variable to hold this as in:
```js
var self = this;
```